cmd/infra/aws/cloudformation/cluster.yaml

```diff
@@ -0,0 +1,488 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Template for HyperShift Cluster Security Elements (Security Groups & IAM)
+
+Parameters:
+  InfrastructureName:
+    AllowedPattern: ^([a-zA-Z][a-zA-Z0-9\-]{0,26})$
+    MaxLength: 27
+    MinLength: 1
+    ConstraintDescription: Infrastructure name must be alphanumeric, start with a letter, and have a maximum of 27 characters.
+    Description: A short, unique cluster ID used to tag cloud resources and identify items owned or used by the cluster.
+    Type: String
+  BaseDomainHostedZoneId:
+    Description: The Route53 public zone ID which is the parent public zone for the cluster, such as Z21IXYZABCZ2A4.
+    Type: String
+  Subdomain:
+    Description: The Route53 delegated zone for the cluster, such as hypershift.example.com. Omit the trailing period.
+    Type: String
+    Default: "hypershift.example.com"
+
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+    - Label:
+        default: "Cluster Information"
+      Parameters:
+      - InfrastructureName
+    ParameterLabels:
+      InfrastructureName:
+        default: "Infrastructure Name"
+
+Mappings:
+  RegionMap:
+    us-east-1:
+      DHCPDomainName: "ec2.internal"
+    us-west-1:
+      DHCPDomainName: "us-west-1.compute.internal"
+
+Resources:
+  OIDCBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Ref InfrastructureName
+      Tags:
+      - Key: !Join [ "", [ "kubernetes.io/cluster/", !Ref InfrastructureName ] ]
+        Value: "owned"
+
+  OIDCProvider:
+    Type: AWS::IAM::OIDCProvider
+    Properties:
+      Tags:
+      - Key: !Join [ "", [ "kubernetes.io/cluster/", !Ref InfrastructureName ] ]
+        Value: "owned"
```

Comment:
I thought we were going to preserve Name tags... it's hard to understand what these are in the console without a name.

Reply:
Sorry, can you remind me what that tag and expected value should be?

Reply:
it's the `Name` tag and for example for the vpc, it's `[INFRA-NAME]-vpc`

Reply:
I went through and added what seem like reasonable Name tags to every tagged resource, PTAL

Reply:
lgtm